iree/compiler/Conversion/LinalgToSPIRV/FoldGPUProcessorIDUses.cpp

```diff
@@ -131,6 +153,61 @@ Optional<int64_t> getProcessorIDUpperBound(gpu::ThreadIdOp threadIDOp) {
   return (*valueIt).getZExtValue();
 }
 
+/// Folds `affin.min` ops over `std.muli` ops with constant operands into
+/// `affine.min` ops.
+///
+/// For example, the following pattern:
+///
+///   %id = hal.interface.workgroup.id[...]
+///   %mul = muli %id %c4
+///   %min = affine.min affine_map<()[s0] -> (9, s0 * -2 + 225)>()[%mul]
+///
+/// Can be folded into:
+///
+///   %min = affine.min affine_map<()[s0] -> (9, s0 * -8 + 225)>()[%id]
+struct FoldConstantMulIntoAffineMin : OpRewritePattern<AffineMinOp> {
```

Comment:
We should not need to do this. We should track down where the `mul` is coming from and make it an affine_apply operation. Then this will get folded automatically.

Reply:
Yeah, I would rather get rid of this pattern. It is very specifically looking for the id multiplied by the constant, etc. If you could track down where the muli instruction is coming from, and change it to an affine apply, you can drop this pattern.

Reply:
That works. Thanks for the suggestion! Sent https://reviews.llvm.org/D98171 for this.